zcash_client_sqlite/src/wallet.rs

```diff
@@ -2338,6 +2350,21 @@ pub(crate) fn put_block(
         ":orchard_action_count": orchard_action_count,
     ])?;
 
+    // If we now have a block corresponding to a received transparent output that
+    // had not been scanned at the time the UTXO was discovered, update the
+    // associated transaction record to refer to that block.
```

Comment:
There's a small data corruption hazard here, in that we're relying exclusively upon the block height to associate the transaction to the block. This is because `CompactBlock` values only contain `CompactTx` entries for transactions that contain shielded inputs or outputs, and the `GetAddressUtxosReply` data does not contain the block hash. As such, it's necessary to ensure that any chain rollback to below the received height causes that height to be set to `NULL`.

Reply:
Add something about that to the comment (maybe referencing `truncate_to_height`).